hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go

```diff
@@ -45,8 +48,67 @@ const (
 	PlaceholderLabel = "hypershift.openshift.io/placeholder"
 
 	autoSizerNamespace = placeholderNamespace
+
+	clusterNamespaceKey = "clusterNamespace"
+	clusterNameKey      = "clusterName"
+	pairLabelKey        = "hypershift.openshift.io/pairlabel"
 )
 
+type PairLabelConfigMapReaper struct {
+	client.Client
+}
+
+func (r *PairLabelConfigMapReaper) SetupWithManager(mgr ctrl.Manager) error {
+	r.Client = mgr.GetClient()
+	tickerChannel := make(chan event.GenericEvent)
+	builder := ctrl.NewControllerManagedBy(mgr).
+		WatchesRawSource(&source.Channel{Source: tickerChannel}, &handler.EnqueueRequestForObject{}).
+		WithOptions(controller.Options{
+			RateLimiter:             workqueue.NewItemExponentialFailureRateLimiter(1*time.Second, 10*time.Second),
+			MaxConcurrentReconciles: 1,
+		}).
+		Named("PairLabelConfigMapReaper")
+
+	go func() {
+		ticker := time.NewTicker(1 * time.Minute)
+		defer ticker.Stop()
+		for range ticker.C {
+			cm := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "ticker", Namespace: "none"}}
+			tickerChannel <- event.GenericEvent{Object: cm}
+		}
+	}()
+	return builder.Complete(r)
+}
+
+func (r *PairLabelConfigMapReaper) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Reconciling")
+	configMapList := &corev1.ConfigMapList{}
+	if err := r.List(ctx, configMapList, client.HasLabels{pairLabelKey}, client.InNamespace(placeholderNamespace)); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to list configmaps: %w", err)
+	}
+	for _, cm := range configMapList.Items {
+		hc := &hyperv1.HostedCluster{}
+		hc.Namespace = cm.Data[clusterNamespaceKey]
```

Comment:
Isn't all this reconcile implementing an ownerRef + garbage collection behaviour, i.e. wouldn't set an ownerRef in the ConfigMap to the HC be enough?

Reply:
I'd actually this reaper to be an "applier" https://github.com/openshift/hypershift/pull/4103/commits/cd2f799f41097b135b6a7f899c3294da3d7d0266#r1618385200 and let reaper be ownerRef builtin behaviour.

Reply:
cross-namespace owner refs [are not allowed](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/), so the HC and configmap would need to be in the same namespace.